src/coreclr/tests/src/CLRTest.CrossGen.targets

```diff
@@ -161,24 +174,41 @@ if defined RunCrossGen (
 )
 REM CrossGen2 Script
 if defined RunCrossGen2 (
-    if defined LargeVersionBubble ( set OptionalArguments=!OptionalArguments! /largeversionbubble)
+    if defined LargeVersionBubble ( set ExtraCrossGen2Args=!ExtraCrossGen2Args! --inputbubble)
     call :TakeLock
     set CrossGen2Status=0
     if not exist "IL" (
         mkdir IL
         if defined CompositeBuildMode (
           copy *.dll IL\
         ) else (
-          copy $(MSBuildProjectName).dll IL\
+          copy *.dll IL\
         )
     )
 
     if defined CompositeBuildMode (
+        set ExtraCrossGen2Args=!ExtraCrossGen2Args! --composite
         set __OutputFile=!scriptPath!\composite-r2r.dll
+        rem In composite mode, treat all dll's in the test folder as rooting inputs
+        set __InputFile=!scriptPath!IL\*.dll
+        call :CompileOneFileCrossgen2
+        IF NOT !CrossGen2Status!==0 goto :DoneCrossgen2Operations
     ) else (
-        set __OutputFile=!scriptPath!\$(MSBuildProjectName).dll
+        for %%I in (!scriptPath!\*.dll) do (
+            set ExtraCrossGen2Args=!ExtraCrossGen2Args! -r:!scriptPath!IL\*.dll
+            set __OutputFile=!scriptPath!\%%~nI.dll
+            set __InputFile=!scriptPath!IL\%%~nI.dll
+            call :CompileOneFileCrossgen2
+            IF NOT !CrossGen2Status!==0 (
+              IF NOT !CrossGen2Status!==2 goto :DoneCrossgen2Operations
```

Comment:
I see the handling of 2 here but not in the bash version.

Reply:
On Unix there aren't any files with the .dll extension other than managed things.